packages/common/src/util/itertools.ts

```diff
@@ -50,3 +50,19 @@ export function partition<T, U>(
   }
   return [first, second];
 }
+
+/**
+ * Returns `true` if the given iterable is empty, `false` otherwise
+ *
+ * From https://github.com/sindresorhus/is-empty-iterable/blob/12d3b4f966170d9d85a2067f5326668d5bb910a0/index.js
+ * @param iterable The iterable to check
+ * @returns `true` if the iterable is empty, `false` otherwise
+ */
+export function isEmptyIterable(iterable: Iterable<unknown>): boolean {
+  for (const _ of iterable) {
+    // eslint-disable-line no-unused-vars, no-unreachable-loop
```

Comment:
`no-unused-vars` is probably not needed

Reply:
Actually appears none of them are needed.  Just copy-pasted this function from a sindresorhus package.  Looks like our lint rules aren't as strict 😅

Reply:
We have made an exception for variable starting with underscore
https://github.com/cursorless-dev/cursorless/blob/b3541bca1b076fd7a95d6bba6b9d094461f98d7f/.eslintrc.json#L30-L31